File: sample/OpenCvSharp.Samples/Samples/BinarizationMethods.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using OpenCvSharp;

// for Binarizer
using OpenCvSharp.Extensions;

namespace OpenCvSharp.Test
{
    /// <summary>
    /// Various Binarization Methods
    /// </summary>
    class BinarizationMethods
    {
        public BinarizationMethods()
        {
            using (IplImage imgSrc = new IplImage(Const.ImageBinarization, LoadMode.GrayScale))
            using (IplImage imgGauss = imgSrc.Clone())
            using (IplImage imgNiblack = new IplImage(imgSrc.Size, BitDepth.U8, 1))
            using (IplImage imgSauvola = new IplImage(imgSrc.Size, BitDepth.U8, 1))
            using (IplImage imgBernsen = new IplImage(imgSrc.Size, BitDepth.U8, 1))
            {
                //Cv.Smooth(imgSrc, imgGauss, SmoothType.Gaussian, 9);
                //Cv.EqualizeHist(imgGauss, imgGauss);

                Stopwatch sw = Stopwatch.StartNew();
                {
                    const int Size = 31;
                    const double K = 0.2;
                    Binarizer.NiblackFast(imgGauss, imgNiblack, Size, K);
                }
                Console.WriteLine("Niblack: {0}ms", sw.ElapsedMilliseconds);
                sw.Reset(); sw.Start();
                {
                    const int Size = 31;
                    const double K = 0.9;
                    const double R = 32;
                    Binarizer.SauvolaFast(imgGauss, imgSauvola, Size, K, R);
                }
                Console.WriteLine("Sauvola: {0}ms", sw.ElapsedMilliseconds); 
                sw.Reset(); sw.Start();
                {
                    const int Size = 31;
                    const byte ContrastMin = 15;
                    const byte BgThreshold = 127;
                    Binarizer.Bernsen(imgGauss, imgBernsen, Size, ContrastMin, BgThreshold);
                }
                Console.WriteLine("bernsen: {0}ms", sw.ElapsedMilliseconds);

                using (new CvWindow("src", imgSrc))
                //using (new CvWindow("gauss", imgGauss))
                using (new CvWindow("niblack", imgNiblack))
                using (new CvWindow("sauvola", imgSauvola))
                using (new CvWindow("bernsen", imgBernsen))
                {
                    Cv.WaitKey();
                }
            }
        }
    }
}

File: sample/CStyleSamplesVB/Samples/BinarizationMethods.vb
Imports System
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Linq
Imports System.Text
Imports OpenCvSharp

' for Binarizer
Imports OpenCvSharp.Extensions

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' Various Binarization Methods
''' </summary>
    Friend Module BinarizationMethods
        Public Sub Start()
        Using imgSrc As New IplImage(FilePath.Image.Binarization, LoadMode.GrayScale), _
             imgGauss As IplImage = imgSrc.Clone(), _
         imgNiblack As New IplImage(imgSrc.Size, BitDepth.U8, 1), _
         imgSauvola As New IplImage(imgSrc.Size, BitDepth.U8, 1), _
         imgBernsen As New IplImage(imgSrc.Size, BitDepth.U8, 1)
            'Cv.Smooth(imgSrc, imgGauss, SmoothType.Gaussian, 9);
            'Cv.EqualizeHist(imgGauss, imgGauss);
            Dim K As Double = 0
            Dim Size As Integer = 0
            Dim sw As Stopwatch = Stopwatch.StartNew()
            Size = 31
            K = 0.2
            Binarizer.NiblackFast(imgGauss, imgNiblack, Size, K)
            Form1.TextBox1.AppendText("Niblack : " & sw.ElapsedMilliseconds.ToString & " ms")
            sw.Reset()
            sw.Start()
            Size = 31
            K = 0.9
            Const R As Double = 32
            Binarizer.SauvolaFast(imgGauss, imgSauvola, Size, K, R)
            Form1.TextBox1.AppendText(Environment.NewLine & "Sauvola: " & sw.ElapsedMilliseconds.ToString & " ms")
            sw.Reset()
            sw.Start()
            Size = 31
            Const ContrastMin As Byte = 15
            Const BgThreshold As Byte = 127
            Binarizer.Bernsen(imgGauss, imgBernsen, Size, ContrastMin, BgThreshold)
            Form1.TextBox1.AppendText(Environment.NewLine & "Bernsen : " & sw.ElapsedMilliseconds.ToString & " ms")

            Using New CvWindow("src", imgSrc)
                Using New CvWindow("niblack", imgNiblack)
                    Using New CvWindow("sauvola", imgSauvola)
                        Using New CvWindow("bernsen", imgBernsen)
                            Cv.WaitKey()
                        End Using
                    End Using
                End Using
            End Using
        End Using
    End Sub
    End Module
' End Namespace
